guides/source/association_basics.md

```diff
@@ -355,2288 +447,2474 @@ class CreateAccountHistories < ActiveRecord::Migration[7.2]
+singularized version of that symbol. For example, given the declaration:
 
 ```ruby
 class Author < ApplicationRecord
-  self.primary_key = [:first_name, :last_name]
-  has_many :books, foreign_key: [:first_name, :last_name]
-end
-
-class Book < ApplicationRecord
-  belongs_to :author, foreign_key: [:author_first_name, :author_last_name]
+  has_many :books
 end
 ```
 
-Accessing a book's author:
-
-```ruby
-author = Author.create!(first_name: "Jane", last_name: "Doe")
-book = author.books.create!(title: "A Cool Book")
+An instance of the `Author` model can have the following methods:
 
-book.reload.author
+```
+books
+books<<(object, ...)
+books.delete(object, ...)
+books.destroy(object, ...)
+books=(objects)
+book_ids
+book_ids=(ids)
+books.clear
+books.empty?
+books.size
+books.find(...)
+books.where(...)
+books.exists?(...)
+books.build(attributes = {}, ...)
+books.create(attributes = {})
+books.create!(attributes = {})
+books.reload
 ```
 
-will use `:first_name` _and_ `:last_name` in the SQL query:
+[`collection<<`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-3C-3C
+[`collection.build`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-build
+[`collection.clear`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-clear
+[`collection.create`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-create
+[`collection.create!`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-create-21
+[`collection.delete`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-delete
+[`collection.destroy`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-destroy
+[`collection.empty?`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-empty-3F
+[`collection.exists?`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F
+[`collection.find`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-find
+[`collection.reload`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-reload
+[`collection.size`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-size
+[`collection.where`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-where
 
-```sql
-SELECT * FROM authors WHERE first_name = 'Jane' AND last_name = 'Doe'
+##### `collection`
+
+The `collection` method returns a Relation of all of the associated objects. If
+there are no associated objects, it returns an empty Relation.
+
+```ruby
+@books = @author.books
 ```
 
-### Self Joins
+##### `collection<<(object, ...)`
```

Comment:
Could we group some of these methods by their use-case? Similar to `belongs_to` and `has_one`?

##### Querying the Association
    collection
    collection_singular_ids
    collection_singular_ids=(ids)
    collection.empty?
    collection.size
    collection.find(...)
    collection.where(...)
    collection.exists?(...)

##### Modifying the Association
    collection<<(object, ...)
    collection=(objects)
    collection.build(attributes = {})
    collection.create(attributes = {})
    collection.create!(attributes = {})
    collection.delete(object, ...)
    collection.destroy(object, ...)
    collection.clear
    collection.reload

Same thing applies to the `has_many :through` association.

Reply:
✅